py/mkrules.mk

```diff
@@ -7,20 +7,19 @@ endif
 # Extra deps that need to happen before object compilation.
 OBJ_EXTRA_ORDER_DEPS =
 
-# QSTR generation uses the same CFLAGS, with these modifications.
-# Note: := to force evalulation immediately.
-QSTR_GEN_CFLAGS := $(CFLAGS)
-QSTR_GEN_CFLAGS += -DNO_QSTR
-QSTR_GEN_CFLAGS += -I$(BUILD)/tmp
-
 ifeq ($(MICROPY_ROM_TEXT_COMPRESSION),1)
 # If compression is enabled, trigger the build of compressed.data.h...
 OBJ_EXTRA_ORDER_DEPS += $(HEADER_BUILD)/compressed.data.h
 # ...and enable the MP_COMPRESSED_ROM_TEXT macro (used by MP_ERROR_TEXT).
-# Note, this doesn't get added to the QSTR_GEN_CFLAGS.
```

Comment:
@jimmo was there a specific reason for not adding the flag to QSTR_GEN_CFLAGS?  In this commit it is now added to it.

I guess it was because of the error including `"genhdr/compressed.data.h"` before it's generated, but that can be fixed by checking for `NO_QSTR` in the C code (see below).

Reply:
Yes, exactly that.

I'm happy with this change though (with some minor comments which I'll add in the relevant places).